### PR TITLE
Posts with ajax

### PR DESCRIPTION
**Pull request by tuananhhoangit**

### CODE REVIEW

**Comment by tuananhhoangit** [comment]
ready

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,18 +1,33 @@
...
+
   def create
-    @post = current_user.posts.build post_params
+    @post = current_user.posts.build(id: params[:id], title: params[:title], content: params[:content])
 
     if @post.save
-      flash[:success] = t ".success"
+      render json: {status: :success, html: render_to_string(@post)}
+    else
+      render json: {status: :error, html: "error"}
```
Comment (huongnguyenmta) [inline app/controllers/posts_controller.rb]: I18n

File: app/assets/javascripts/posts.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+          alert('fail');
+        });
+      return false;
+    }
+  });
+
+  $('body').on('click', '.delete-post', function(event){
+    event.preventDefault();
+    var self = $(this);
+    var r = confirm('You sure');
```
Comment (huongnguyenmta) [inline app/assets/javascripts/posts.js]: I18n

File: app/assets/javascripts/posts.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    if (self.attr('action') == 'edit') {
+      $.ajax({
+        url: '/posts/'+post_id,
+        type: 'put',
+        dataType: 'json',
+        data: {id: post_id, title: new_title, content: new_content},
+      })
+        .done(function(response) {
+          if (response.status == 'success') {
+            $('#post-'+post_id).find('h3').text(new_title);
```
Comment (huongnguyenmta) [inline app/assets/javascripts/posts.js]: `$('#post-' + post_id)`

File: app/assets/javascripts/posts.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+            $('#post-'+post_id).find('p').text(new_content);
```
Comment (huongnguyenmta) [inline app/assets/javascripts/posts.js]: `$('#post-' + post_id)`

File: app/assets/javascripts/posts.js [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+          }
+        })
+        .fail(function() {
+          alert('fail');
+        });
+      return false;
+    } else {
+      $.ajax({
+        url: '/posts',
+        type: 'post',
```
Comment (huongnguyenmta) [inline app/assets/javascripts/posts.js]: `type: 'POST'`

File: app/assets/javascripts/posts.js [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+  $('body').on('click', '.post', function(event){
+    event.preventDefault();
+    var self = $(this);
+    var new_title = $('.title').val();
+    var new_content = $('.content').val();
+    var post_id = self.attr('post_id');
+    if (self.attr('action') == 'edit') {
+      $.ajax({
+        url: '/posts/'+post_id,
+        type: 'put',
```
Comment (huongnguyenmta) [inline app/assets/javascripts/posts.js]: `type: 'PUT'`

File: app/controllers/posts_controller.rb [hunk 7]
```diff
@@ -1,18 +1,33 @@
...
+      render json: {status: :success}
+    else
+      render json: {status: :error, html: "error"}
+    end
+  end
 
-  def post_params
-    params.require(:post).permit :id, :title, :content, :picture
+  def load_post
+    @post = Post.find_by id: params[:id]
```
Comment (huongnguyenmta) [inline app/controllers/posts_controller.rb]: check case not found

**Comment by huongnguyenmta** [comment]
commented

**Comment by tuananhhoangit** [comment]
ready

**Pull request merged**